### PR TITLE
Upgrade linter to golangci-lint, bump Go to 1.12 for Travis

### PR DESCRIPTION
**Pull request by AlfredBroda**
Remove deprecated gometalinter

### CODE REVIEW

File: consul/consul.go [hunk 1]
```diff
@@ -46,23 +47,23 @@ func (c *Consul) getServicesUsingProviderWithRetriesOnAgentFailure(provide Servi
...
 			if failures := agent.IncFailures(); agent != localAgent && failures > c.config.AgentFailuresTolerance {
 				log.WithField("Address", agent.IP).WithField("Failures", failures).
 					Warn("Removing agent due to too many failures")
 				c.agents.RemoveAgent(agent.IP)
 			}
-		} else {
-			agent.ClearFailures()
-			return services, nil
 		}
+		agent.ClearFailures()
```
Comment (pbetkier) [inline consul/consul.go]: Missing else? You changed the logic here.

File: consul/consul.go [hunk 2]
```diff
@@ -162,13 +163,12 @@ func (c *Consul) Register(task *apps.Task, app *apps.App) error {
 	metrics.Time("consul.register", func() { err = c.registerMultipleServices(services) })
 	if err != nil {
 		metrics.Mark("consul.register.error")
-	} else {
-		metrics.Mark("consul.register.success")
 	}
+	metrics.Mark("consul.register.success")
```
Comment (pbetkier) [inline consul/consul.go]: logic changed (missing else)

File: sse/sse_handler.go [hunk 3]
```diff
@@ -5,22 +5,23 @@ import (
...
+	log "github.com/sirupsen/logrus"
+
 	"github.com/allegro/marathon-consul/events"
 	"github.com/allegro/marathon-consul/marathon"
 	"github.com/allegro/marathon-consul/metrics"
-	log "github.com/sirupsen/logrus"
 )
 
-// SSEHandler defines handler for marathon event stream, opening and closing
+// HandlerSSE defines handler for marathon event stream, opening and closing
```
Comment (pbetkier) [inline sse/sse_handler.go]: why this rename?

Comment (AlfredBroda) [reply sse/sse_handler.go]: Linter error: `sse/sse_handler.go:17:6: type name will be used as sse.SSEHandler by other packages, and that stutters; consider calling this Handler (golint)`
Unfortunately Handler is already taken by a func in `sse/sse.go`

File: consul/consul.go (hunk 2 above)
Comment (AlfredBroda) [reply consul/consul.go]: Reverted

File: consul/consul.go (hunk 1 above)
Comment (AlfredBroda) [reply consul/consul.go]: Reverted

File: consul/consul.go [hunk 4]
```diff
@@ -359,12 +361,11 @@ func marathonToConsulCheck(task *apps.Task, check apps.HealthCheck, serviceAddre
 			parsedURL.Host = fmt.Sprintf("%s:%d", serviceAddress, port)
 			consulCheck.HTTP = parsedURL.String()
 			return consulCheck
-		} else {
-			log.WithError(err).
-				WithField("Id", task.AppID.String()).
-				WithField("Address", serviceAddress).
-				Warnf("Could not parse provided path: %s", path)
 		}
+		log.WithError(err).
```
Comment (ojagodzinski) [inline consul/consul.go]: Missing else?

Comment (AlfredBroda) [reply consul/consul.go]: No, the return in the if block makes it unnecessary, the inter complained about this.

**Pull request merged**